torch/_prims/__init__.py

```diff
@@ -694,7 +694,10 @@ def _clone_meta(
         )
 
     # memory_format == torch.preserve_format
-    strides = utils.compute_elementwise_output_strides(input)
+    if torch._prims_common.is_non_overlapping_and_dense(input):
+        strides = input.stride()
+    else:
+        strides = input.contiguous().stride()
```

Comment:
I'm not completely sure this is correct. Also, assuming that it is correct, we should update utils.compute_elementwise_output_strides -- other operators run into the same problem.

This is the function we use to compute strides when the input is not "non_overlapping_and_dense": https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/TensorIterator.cpp#L1276.

If we think it is exactly `input.contiguous.stride()`, then we should add some unittests to check that

Reply:
EDIT: .contiguous calls exactly that function so it is actually the correct thing. So I think my ask is that we should update `utils.compute_elementwise_output_strides` to call 
```py
if torch._prims_common.is_non_overlapping_and_dense(input):
    strides = input.stride()
else:
    strides = input.contiguous().stride()
```